### DISCUSSION TITLE
How to change plugin settings in the init function (Scala statusBarProvide)

### DISCUSSION THREAD

**Discussion by fikovnik** [LazyVim/LazyVim#2962] [Q&A]
(Thank you for including support for Scala!)

I was wondering what is the right way to change the init options, concretely the [line 23](https://github.com/LazyVim/LazyVim/blob/97480dc5d2dbb717b45a351e0b04835f138a9094/lua/lazyvim/plugins/extras/lang/scala.lua#L23) from "on" to "off" without copying the entire file:

```lua
...
{
    "scalameta/nvim-metals",
    ...    
    init = function()
      local metals_config = require("metals").bare_config()
      metals_config.init_options.statusBarProvider = "on"   -- how to set this to "off"?
...
```

Setting it to "off" makes it compatible with the general LSP status plugins like [fidget](https://github.com/j-hui/fidget.nvim) which is super useful to get feedback about loading (Scala LSP metals takes quite some time to get ready).

**Comment by dpetka2001** [comment]
You can't the way it's structured at the moment. You would have to rewrite the whole config. Here's an example from the [plugin's docs](https://github.com/scalameta/nvim-metals?tab=readme-ov-file#absolute-minimal-example-using-lazynvim). I don't know who made the PR for scala support, but it looks like `opts` should have been used instead of `init`, so that the users could then make custom changes in their personal configuration without rewriting the whole thing. 

I don't do Scala myself and can't test this, but if you manage to make it work with the example from the plugin's docs that I linked, then maybe a change to the Scala Extra would be appropriate.

**Reply by dpetka2001** [reply]
Also the following [lines](https://github.com/LazyVim/LazyVim/blob/97480dc5d2dbb717b45a351e0b04835f138a9094/lua/lazyvim/plugins/extras/lang/scala.lua#L6-L13) are wrong, because `lazy.nvim` doesn't have a `requires` Plugin spec key, rather a `dependencies`. And the use of `vsnip` engine is a bit opinionated in my personal opinion, since LazyVim already uses Luasnip and this should be left to be added in the user personal configuration instead.

PS: You could also take a look at the github repo link that is mentioned at the top of the file for a more complete configuration than the minimal one mentioned in the README.
